tests/test-surround-view.cpp

```diff
@@ -320,22 +329,59 @@ remap_topview_buf (const SmartPtr<SVStream> &stitch, const SmartPtr<SVStream> &t
     return XCAM_RETURN_NO_ERROR;
 }
 
+static XCamReturn
+create_cubemap_mapper (
+    const SmartPtr<Stitcher> &stitcher, const SmartPtr<SVStream> &stitch,
+    const SmartPtr<SVStream> &cubemap, SVModule module)
+{
+    CubeMapModel cubemap_model (stitch->get_width (), stitch->get_height ());
+    CubeMapModel::PointMap points;
+
+    cubemap_model.get_cubemap_rect_map (points, cubemap->get_width (), cubemap->get_height ());
+    SmartPtr<GeoMapper> mapper;
+    if (module == SVModuleSoft) {
+        mapper = GeoMapper::create_soft_geo_mapper ();
+    } else if (module == SVModuleGLES) {
+#if HAVE_GLES
+        mapper = GeoMapper::create_gl_geo_mapper ();
+#endif
+    } else if (module == SVModuleVulkan) {
+#if HAVE_VULKAN
+        SmartPtr<VKDevice> dev = stitch->get_vk_device ();
+        XCAM_ASSERT (dev.ptr ());
+        mapper = GeoMapper::create_vk_geo_mapper (dev, "cubemap-map");
+#endif
+    }
+    XCAM_ASSERT (mapper.ptr ());
+
+    mapper->set_output_size (cubemap->get_width (), cubemap->get_height ());
+    mapper->set_lookup_table (points.data (), cubemap->get_width (), cubemap->get_height ());
+    cubemap->set_mapper (mapper);
+
+    return XCAM_RETURN_NO_ERROR;
+}
+
 static void
 write_image (
     const SmartPtr<Stitcher> &stitcher,
     const SVStreams &ins, const SVStreams &outs,
-    bool save_output, bool save_topview)
+    const SVOutConfig &out_config)
 {
     static uint32_t frame_num = 0;
 
     write_in_image (stitcher, ins, frame_num);
 
-    if (save_output)
-        write_out_image (outs[IdxStitch], frame_num);
+    if (out_config.save_output)
+        write_out_image (outs[out_config.stitch_index], frame_num);
+
+    if (out_config.save_topview) {
+        remap_topview_buf (outs[out_config.stitch_index], outs[out_config.topview_index]);
+        write_out_image (outs[out_config.topview_index], frame_num);
+    }
 
-    if (save_topview) {
-        remap_topview_buf (outs[IdxStitch], outs[IdxTopView]);
-        write_out_image (outs[IdxTopView], frame_num);
+    if (out_config.save_cubemap) {
+        remap_topview_buf (outs[out_config.stitch_index], outs[out_config.cubemap_index]);
```

Comment:
can we change this "remap_topview_buf" function name to "remap_buf"? since this function is used by topview & cubmap remap.

Reply:
Makes sense. Renamed this function. Thanks!